### PR TITLE
Problem: memiavl is not integrated

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- integrate memiavl feature

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - [#975](https://github.com/crypto-org-chain/chain-main/pull/975) Integrate local state-sync commands.
 - [#988](https://github.com/crypto-org-chain/chain-main/pull/988) Release testnet binaries together.
+- [#]() Integrate memiavl feature.
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
- [#989](https://github.com/crypto-org-chain/chain-main/pull/989) Integrate memiavl feature.
```

File: app/config/toml.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package config
+
+// DefaultConfigTemplate defines the configuration template for the EVM RPC configuration
+const DefaultConfigTemplate = `
+###############################################################################
+###                             Cronos Configuration                        ###
+###############################################################################
```
Comment (yihuang) [inline app/config/toml.go:7]:
```suggestion
// DefaultConfigTemplate defines the configuration template for the memiavl configuration
const DefaultConfigTemplate = `
###############################################################################
###                             MemIAVL Configuration                        ###
###############################################################################
```

**Review by mmsqe — APPROVED** [review]
LGTM, just not sure if need add [grocksdb_clean_link](https://github.com/crypto-org-chain/cronos/blob/main/Makefile#L57)

**Comment by yihuang** [comment]
> LGTM, just not sure if need add [grocksdb_clean_link](https://github.com/crypto-org-chain/cronos/blob/main/Makefile#L57)

it depends on user's local rocksdb library build, if static linking then no need, so it's alright in Makefile I think.

File: go.mod [hunk 3]
```diff
@@ -194,6 +195,7 @@ replace (
 	// use cosmos fork of keyring
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
 	github.com/cometbft/cometbft-db => github.com/crypto-org-chain/cometbft-db v0.0.0-20230412133340-ac70df4b45f6
+	github.com/crypto-org-chain/cronos/memiavl => github.com/yihuang/cronos/memiavl v0.0.0-20230615081901-24c1ef8782fd
```
Comment (yihuang) [inline go.mod:198]: wait for cronos merge

**Pull request merged**